tests/storage/test_factory.py

```diff
@@ -0,0 +1,34 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+
+import re
+from unittest import TestCase
+
+from storage import CommonStorage, S3Storage, StorageFactory
+
+
+class TestStorageFactory(TestCase):
+    def test_create(self) -> None:
+        with self.subTest("create s3 storage success"):
```

Comment:
I didn't know this way of defining subtests in python, nice :slightly_smiling_face: